Code/GraphMol/FileParsers/Mol2FileParser.cpp

```diff
@@ -902,48 +901,32 @@ RWMol *Mol2DataStreamToMol(std::istream *inStream, bool sanitize, bool removeHs,
   res->setProp("_TriposChargeType", tempStr);
   // stop here since we don't support anything else from the MOLECULE block
   INT_VECT idxCorresp(nAtoms, -1);
-  inStream->seekg(atomStart, std::ios::beg);
-  try {
-    ParseMol2AtomBlock(inStream, res, nAtoms, idxCorresp);
-  } catch (const FileParseException &e) {
-    delete res;
-    throw e;
-  }
+  inStream.seekg(atomStart, std::ios::beg);
+  ParseMol2AtomBlock(&inStream, res.get(), nAtoms, idxCorresp);
   if (nBonds) {
     // stop here since we don't support anything else from the MOLECULE block
-    inStream->seekg(bondStart, std::ios::beg);
-    try {
-      ParseMol2BondBlock(inStream, res, nBonds, idxCorresp);
-    } catch (const FileParseException &e) {
-      delete res;
-      throw e;
-    }
+    inStream.seekg(bondStart, std::ios::beg);
+    ParseMol2BondBlock(&inStream, res.get(), nBonds, idxCorresp);
   }
 
   if (!chargeStart) {
     bool molFixed;
-    if (cleanupSubstructures) {
-      molFixed = cleanUpMol2Substructures(res);
+    if (params.cleanupSubstructures) {
+      molFixed = cleanUpMol2Substructures(res.get());
     } else {
       molFixed = true;
     }
 
     if (!molFixed) {
-      delete res;
       return nullptr;
     }
 
     // mol2 format does not support formal charge information, hence we need to
     // guess it based on default and explicit valences
-    guessFormalCharges(res);
+    guessFormalCharges(res.get());
   } else {
-    inStream->seekg(chargeStart, std::ios::beg);
-    try {
-      readFormalChargesFromAttr(inStream, res);
-    } catch (const FileParseException &e) {
-      delete res;
-      throw e;
-    }
+    inStream.seekg(chargeStart, std::ios::beg);
+    readFormalChargesFromAttr(&inStream, res.get());
```

Comment:
Should we make a v2 readFormalCharges that takes a reference to a mol not a pointer?

```
readformalChargesFromAttr(&inStream, *res);
```

Reply:
Sure, I would do that if writing things from scratch, but going back and updating stuff that's not in the public API doesn't feel like a super productive use of time.

Reply:
Agreed.